### DISCUSSION TITLE
GitHub story

### DISCUSSION THREAD

**Discussion by wqweto** [twinbasic/twinbasic#1165] [General]
I've been following @fafalone's [EventTrace effort](https://github.com/fafalone/EventTrace) here on github and being unable to browse the sources in the `twinproj` amalgamated file is very unfortunate because even VB6 sources there are browsable although interspersed with extra attributes (which VBIDE conveniently hides) and not being utf-8 (which leads to ANSI codepage being wrongly auto-detected in 99% of the cases).

I was wondering if something can be done to improve "visibility" of TB sources in public repositories or this is an unfortunate consequence of early design decisions that cannot be undone at this point in time with project nearing v1.0.

cheers,
&lt;/wqw>

**Comment by WaynePhillipsEA** [comment]
Allowing to use external files is still planned (thus then the twinproj file can act much like a VBP).

For now, you should look at the project settings `Project: Export After Save` and `Project: Export Path`.  By setting those two options, tB will then auto-export all the individual project files each time you save the project.    There is also an `Import from folder` option on the New Project dialog to do the reverse.

**Comment by fafalone** [comment]
I added the exported .twin files... I don't know how to get github to apply the VBA syntax highlighting though. I put a .gitattributes file in the folder and a few things I've tried haven't worked. I don't even think it's uploading the file.

**Comment by wqweto** [comment]
I don’t think you can configure github highlighter by file extension from any dot file in the repo as its a setting in Linguist component they are using.

Currently it does not highlight VB6 .ctl files from user-controls and probably a lot of other extensions (like .pag or .dob) are missing too.

We could prepare a Pull Request which includes both VB6 fixes and “smuggle” .twin extension in there too. This is what V Lang did, they hijacked golang highlighter :))

**Comment by fafalone** [comment]
https://github.com/github/linguist/issues/1792

That was closed as implemented, suggesting you should be able to buy defining .twin as an alias for a VB extension... I can't get it though.

**Reply by wqweto** [reply]
Wow, this is good news and it's really working!

I just added `*.ctl linguist-language=vba` to [my repo's `.gitattributes`](https://github.com/Unicontsoft/UcsFiscalPrinters/blob/master/.gitattributes) and it started having [AlphaBlendLabel.ctl](https://github.com/Unicontsoft/UcsFiscalPrinters/blob/master/src/UcsFPHub/Shared/AlphaBlendLabel.ctl) syntax highlighted in source browse just fine.

I'm just wondering for `.twin` files which one will highlight better -- `vba` or `vbscript` highlighter? More testing needed.

**Comment by fafalone** [comment]
I don't understand how to add such a file. I get a 404 at that URL and if I try uploading it, it says 'This file is hidden' and then doesn't upload anything. And yes I've searched; not a single one of the ways you're supposed to be able to do it are working for me. 

Edit: Nevermind, finally got it by using 'add new file' and manually creating one.

**Reply by wqweto** [reply]
Btw, you can try setting .tbform to json too with something like
```
*.twin linguist-language=vba
*.tbform linguist-language=json
```
